PHPfiles/ProductCalendar.php

```diff
@@ -38,12 +50,15 @@ function renderCalendar() {
                 $columnNumber = 0;
             }
             if ($productId === ProductCalendar::NoLunchProductId) {
-                $dateString = date_format(new DateTime($productDate), "M d, Y");
-                HtmlHelpers::writeTableCell("$dateString <p><i>No lunch</i></p>", ThemeConstants::TableCellNothingToOrderStyle);
+                ProductCalendar::renderNoLunchTableCell($productDate);
             } elseif ($productId === ProductCalendar::NoDetailsProductId) {
-                HtmlHelpers::writeTableCell("&nbsp;", ThemeConstants::TableCellNothingToOrderStyle);                
+                ProductCalendar::renderNoProductDetailsTableCell();
             } else {
-                display_meal($productId);
+                $productDetails = wc_get_product($productId);
```

Comment:
Love your naming for these variables and how you split it out.  much easier to understand and each individual piece is it's own function

Reply:
Thanks!  Also, we're calling wc_get_product only once for the parent product id.